src/common/com/intellij/plugins/haxe/model/HaxePackageModel.java

```diff
@@ -125,9 +125,23 @@ public HaxeFileModel getFileModel(String fileName) {
   protected HaxeFile getFile(String fileName) {
     PsiDirectory directory = root.access(path);
     if (directory != null && directory.isValid()) {
-      PsiFile file = directory.findFile(fileName + ".hx");
-      if (file != null && file.isValid() && file instanceof HaxeFile) {
-        return (HaxeFile)file;
+      PsiFile file;
+      try {
+        String fName = fileName + ".hx";
+        //TODO @ Eric.  What could be the source of a request for "Unknown.hx" ?
+        if(fName.equals("Unknown.hx")){
```

Comment:
BTW, there is nothing in the language that says you can't have a class called `Unknown` and an `Unknown.hx` file to go with it.  So, just returning null when we see that name is going to create other bugs.

Perhaps we should rename our internal `unknown` class to `__unknown__`?  The double leading underscore is soft-reserved for compiler internals anyway.  Unfortunately, I'm sure that it will pop out to the UI somewhere...

Reply:
I agree...I didnt expect my little hack to be permanent.. Shall I try to find that magic type and rename it as you recommended?